nilearn/_utils/niimg.py

```diff
@@ -127,14 +127,16 @@ def load_niimg(niimg, dtype=None):
     dtype = _get_target_dtype(_get_data(niimg).dtype, dtype)
 
     if dtype is not None:
-        # Copyheader and set dtype in header if header exists
-        if niimg.header is not None:
+        # Copy header and set dtype in header if header exists
+        try:  # Is header iterable
+            'something' in niimg.header
+        except TypeError:  # incompatible with new_img_like(copy_header=True).
             niimg = new_img_like(niimg, _get_data(niimg).astype(dtype),
-                                niimg.affine, copy_header=True)
-            niimg.header.set_data_dtype(dtype)        
```

Comment:
That's a very hacky solution. From what I read here, it seems that we are no longer keeping the header under the new versions nibabel, aren't we? That would be a problem.

Reply:
This comment references the code in the master branch, not the PR. Was that accidental or are you saying the `master` code` is hacky? It was merged in PR #1872

Reply:
This was accidental: I just checked, and the code in master does copy the header when it's not none.